bin/stringmerge

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env python3
+
+'''
+Merge two existing problems.
+'''
+
+import sys
+import argparse
+
+from stringfuzz.ast import ExpressionNode, SortNode, IdentifierNode, FunctionDeclarationNode, SortedVarNode
+from stringfuzz.ast_walker import ASTWalker
+from stringfuzz.constants import LANGUAGES, SMT_25_STRING
+from stringfuzz.generator import generate
+from stringfuzz.parser import parse, ParsingError
+from stringfuzz.smt import smt_string_logic, smt_sat
+
+# defaults
+DEFAULT_RENAME_IDS = False
+
+# constants
+DECLARE_CONST = "declare-const"
+
+# helper functions 
+def assertion(expr):
+    if isinstance(expr, ExpressionNode):
+        symbol = expr.symbol
+        if isinstance(symbol, IdentifierNode) and symbol.name == "assert":
+            return True
+    return False
+
+def declaration(expr):
+    if isinstance(expr, (FunctionDeclarationNode, SortedVarNode)):
+        return True
+    if isinstance(expr, ExpressionNode):
+        symbol = expr.symbol
+        if isinstance(symbol, IdentifierNode) and symbol.name == DECLARE_CONST:
+            return True
+    return False
+
+def sort(expr):
+    return isinstance(expr, SortNode)
+
+# helper class
+class RenameIDWalker(ASTWalker):
+    def __init__(self, ast, suffix):
+        super(RenameIDWalker, self).__init__(ast)
+        self.suffix = str(suffix)
+
+    def exit_identifier(self, identifier, parent):
+        identifier.name += self.suffix
+
+# most important function
+def merge(asts, rename_ids):
+    if rename_ids:
+        for i in range(len(asts)):
+            asts[i] = RenameIDWalker(asts[i], i).walk()
+
+    flattened    = [expr for ast in asts for expr in ast]
+    combined     = set(flattened)
+    declarations = list(filter(declaration, combined))
+    sorts        = list(filter(sort, combined))
+    assertions   = list(filter(assertion, combined))
+    transformed  = [smt_string_logic()] + declarations + sorts + assertions + [smt_sat()]
+    return transformed
+
+# entry point
+
+def main():
+
+    # create arg parser
+    global_parser = argparse.ArgumentParser(description='SMTLIB 2.* problem merger.')
+    global_parser.add_argument(
+        'files',
+        nargs    = '+',
+        metavar  = 'F',
```

Comment:
Right now the help text shows `F [F ...]`. It might be more readable to call this `file`.